experiments/utils.py

```diff
@@ -141,6 +142,7 @@ def get_alternative(self, experiment_name, request=None):
         if experiment and not disabled:
             if experiment.is_displaying_alternatives():
                 alternative = self._get_enrollment(experiment)
+                self.active_experiment = experiment_name
```

Comment:
We should do the same for `_enroll()` method, it returns a chosen alternative as well.